Mathlib/CategoryTheory/Distributive/Monoidal.lean

```diff
@@ -0,0 +1,321 @@
+`coprod.inl : Y ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)`. -/
+@[reassoc (attr := simp)]
+lemma whisker_inl_inv_rightDistrib {X Y Z : C} :
+    (coprod.inl ▷ X) ≫ (∂R X Y Z).inv = coprod.inl := by
+  apply (cancel_iso_hom_right _ _ (∂R X Y Z)).mp
+  rw [assoc, Iso.inv_hom_id, comp_id, coprod_inl_rightDistrib_hom]
+
+/-- The composite of `(coprod.inr ▷ X) : Z ⊗ X ⟶ (Y ⨿ Z) ⊗ X` and
+`(∂R X Y Z).inv :  (Y ⨿ Z) ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)` is equal to the right coprojection
+`coprod.inr : Z ⊗ X ⟶ (Y ⊗ X) ⨿ (Z ⊗ X)`. -/
+@[reassoc (attr := simp)]
+lemma whisker_inr_inv_rightDistrib {X Y Z : C} :
+    (coprod.inr ▷ X) ≫ (∂R X Y Z).inv = coprod.inr := by
+  apply (cancel_iso_hom_right _ _ (∂R X Y Z)).mp
+  rw [assoc, Iso.inv_hom_id, comp_id, coprod_inr_rightDistrib_hom]
+
+end IsMonoidalRightDistrib
+
+/-- In a symmetric monoidal category, the left distributivity is equal to
+the right distributivity up to braiding isomorphisms. -/
+@[simp]
+lemma SymmetricCategory.leftDistrib_braiding [SymmetricCategory C] {X Y Z : C} :
+    (coprodComparison (tensorLeft X) Y Z) ≫ (β_ X (Y ⨿ Z)).hom =
+    (coprod.map (β_ X Y).hom (β_ X Z).hom) ≫ (coprodComparison (tensorRight X) Y Z) := by
+  simp [coprodComparison]
+
+
+/-- In a symmetric monoidal category, the right distributivity is equal to
+the left distributivity up to braiding isomorphisms. -/
+@[simp]
+lemma SymmetricCategory.rightDistrib_braiding [SymmetricCategory C] {X Y Z : C} :
+    (coprodComparison (tensorRight X) Y Z) ≫ (β_ (Y ⨿ Z) X).hom =
+    (coprod.map (β_ Y X).hom (β_ Z X).hom) ≫ (coprodComparison (tensorLeft X) Y Z) := by
+  simp [coprodComparison]
+
+/-- A left distributive symmetric monoidal category is right distributive. -/
+instance SymmetricCategory.isMonoidalRightDistrib_of_isMonoidalLeftDistrib
+    [SymmetricCategory C] [IsMonoidalLeftDistrib C] : IsMonoidalRightDistrib C where
+  preservesBinaryCoproducts_tensorRight X :=
+    preservesColimitsOfShape_of_natIso (SymmetricCategory.tensorLeftIsoTensorRight X)
+
+/-- A left distributive symmetric monoidal category is distributive. -/
+instance SymmetricCategory.isMonoidalDistrib_of_isMonoidalLeftDistrib
+    [SymmetricCategory C] [IsMonoidalLeftDistrib C] : IsMonoidalDistrib C where
+
+/-- The right distributivity isomorphism of the a left distributive symmetric monoidal category
+is given by `(β_ (Y ⨿ Z) X).hom ≫ (∂L X Y Z).inv ≫ (coprod.map (β_ X Y).hom (β_ X Z).hom)`. -/
+@[simp]
+lemma SymmetricCategory.rightDistrib_of_leftDistrib
+    [SymmetricCategory C] [IsMonoidalLeftDistrib C] {X Y Z : C} :
+    ∂R X Y Z = (coprod.mapIso (β_ Y X) (β_ Z X)) ≪≫ (∂L X Y Z) ≪≫ (β_ X (Y ⨿ Z)) := by
+  ext
+  · simp
+  · simp
+
+/-- A closed monoidal category is left distributive. -/
+instance MonoidalClosed.isMonoidalLeftDistrib [MonoidalClosed C] :
+    IsMonoidalLeftDistrib C where
+  preservesBinaryCoproducts_tensorLeft X := {
+    preservesColimit := ((ihom.adjunction X).leftAdjoint_preservesColimits).1.1
+  }
```

Comment:
```suggestion
  preservesBinaryCoproducts_tensorLeft X := by
    have := (ihom.adjunction X).isLeftAdjoint
    infer_instance
```
It would be useful to add `IsLeftAdjoint` instances for closed monoidal categories, so that the line with `have := ...` would become unnecessary.

Reply:
sounds good, added the following instance to `Closed.Monoidal`: 

```
instance : (tensorLeft A).IsLeftAdjoint  :=
  (ihom.adjunction A).isLeftAdjoint
```